shell/list/node.vue

```diff
@@ -41,30 +40,30 @@ export default {
   },
 
   async fetch() {
-    const hash = { kubeNodes: this.$fetchType(NODE) };
+    this.$initializeFetchData(this.resource);
 
     this.canViewPods = this.$store.getters[`cluster/schemaFor`](POD);
 
     if (this.$store.getters[`management/schemaFor`](MANAGEMENT.NODE)) {
       // Required for Drain/Cordon action
-      hash.normanNodes = this.$store.dispatch('rancher/findAll', { type: NORMAN.NODE });
+      this.$fetchType(NORMAN.NODE, [], 'rancher');
     }
 
     if (this.$store.getters[`rancher/schemaFor`](NORMAN.NODE)) {
-      hash.mgmtNodes = this.$store.dispatch('management/findAll', { type: MANAGEMENT.NODE });
+      this.$fetchType(MANAGEMENT.NODE, [], 'management');
     }
 
     if (this.$store.getters[`management/schemaFor`](CAPI.MACHINE)) {
       // Required for ssh / download key actions
-      hash.machines = this.$store.dispatch('management/findAll', { type: CAPI.MACHINE });
+      this.$fetchType(CAPI.MACHINE, [], 'management');
     }
 
     if (this.canViewPods) {
-      // Used for running pods metrics - we don't need to block on this to show the list of nodes
-      this.$store.dispatch('cluster/findAll', { type: POD });
+      // Used for running pods metrics
+      this.$fetchType(POD);
     }
 
-    await allHash(hash);
+    await this.$fetchType(this.resource);
```

Comment:
Correct, but after my tests I don't see any secondary effects here, afaik.

With the `hash` they are all done in parallel but all of it only resolves when the last request is done.

With the new approach we are keeping the parallel and the UI should auto-update with the correct data.

This can be optimised, for sure, but I don't really know which requests are the ones worth blocking to minimize DOM updates vs the time spent without content on the screen. Delicate balance here. What do you think?